$NetBSD$

Apply upstream patch aaff080449ebcb66553ebf5b512f737b6191f339 to fix
build with gcc 4.6+ and clang. Via Al Viro.

--- parse.h~	2007-05-01 18:30:58.000000000 +0000
+++ parse.h
@@ -35,10 +35,6 @@ struct statement {
 		struct /* declaration */ {
 			struct symbol_list *declaration;
 		};
-		struct /* label_arg */ {
-			struct symbol *label;
-			struct statement *label_statement;
-		};
 		struct {
 			struct expression *expression;
 			struct expression *context;
